Source/ablastr/parallelization/MPIInitHelpers.cpp

```diff
@@ -40,6 +48,16 @@ namespace ablastr::parallelization
     std::pair< int, int >
     mpi_init (int argc, char* argv[])
     {
+        // OLCFDEV-1655: Segfault during MPI_Init & in PMI_Allgather
+        // https://docs.olcf.ornl.gov/systems/crusher_quick_start_guide.html#olcfdev-1655-occasional-seg-fault-during-mpi-init
+#if defined(AMREX_USE_HIP) && defined(AMREX_USE_MPI)
+        hipError_t hip_ok = hipInit(0);
+        if (hip_ok != hipSuccess) {
+            std::cerr << "hipInit failed with error code " << hip_ok << "! Aborting now.\n";
```

Comment:
Is there a reason for not using `ABLASTR_ALWAYS_ASSERT_WITH_MESSAGE` here?

Reply:
Yes: anything calling into AMReX functions cannot be used before AMReX is initialized.